staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -79,6 +81,8 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 	switch {
 	case ok && a.successTTL > 0:
 		a.cache.set(key, &cacheRecord{resp: resp, ok: ok, err: err}, a.successTTL)
+	case !a.cacheErrs && err != nil:
+		return nil, false, err
 	case !ok && a.failureTTL > 0:
```

Comment:
an early return in the middle of the switch reads oddly to me. would it be clearer to conditionalize the second case?

```
case !ok && a.failureTTL > 0 && (a.cacheErrs || err == nil):
  ...
```